actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -33,6 +33,7 @@ def generate(name, options, path_parameters, parameterize = nil)
           defaults       = route.defaults
           required_parts = route.required_parts
           parameterized_parts.delete_if do |key, value|
+            next if defaults[key].nil?
```

Comment:
Thanks! action patch tests pass with that for me. seems good, somehow removing the bangs makes it okay  to be on one line (for me). I pushed this change. Will wait on the whole suite to run again to be sure.